lib/puppet/provider/package/gem.rb

```diff
@@ -17,6 +19,9 @@
 
   has_feature :versionable, :install_options, :uninstall_options, :targetable, :version_ranges
 
+  GEM_VERSION       = Puppet::Util::Package::Version::Gem
+  GEM_VERSION_RANGE = Puppet::Util::Package::Version::Range
+
```

Comment:
These cause warnings for some reason:
> /Users/luchi/repos/puppet/lib/puppet/provider/package/gem.rb:22: warning: already initialized constant GEM_VERSION
/Users/luchi/repos/puppet/lib/puppet/provider/package/gem.rb:22: warning: previous definition of GEM_VERSION was here
/Users/luchi/repos/puppet/lib/puppet/provider/package/gem.rb:23: warning: already initialized constant GEM_VERSION_RANGE
/Users/luchi/repos/puppet/lib/puppet/provider/package/gem.rb:23: warning: previous definition of GEM_VERSION_RANGE was here

Tried with different names but that doesn't help. It acts like the file is being loaded twice. 
`pry-byebug` got me to `puppet/lib/puppet/util/autoload.rb:80` but I couldn't figure the root cause yet. @puppetlabs/night-s-watch ideas are very welcomed 😅

Reply:
After discussing about this, it was decided to remove these variables to avoid the warnings and track the found issue in [PUP-10390](https://tickets.puppetlabs.com/browse/PUP-10390).